### PR TITLE
New Release: 0.3.0

### PR DESCRIPTION
**Pull request by D-o-d-o-x**
This PR tests our new CI release setup, aiming to push a release to PyPI when merged. Such PRs onto release should be blocked when either no version tag is detected or a mismatch between the tag and the version found in pyproject.toml is found. This ensures our version numbers are always in sync.

Content wise, it combines the latest from our master branch with the versioning improvements made in PR #96.

This PR should only be merged after PR #96 was merged into master.

Basically, we're testing to make sure our new release process works as intended. Let's see how it goes.

### CODE REVIEW

**Comment by D-o-d-o-x** [comment]
Version validation seems to work.

If the commit tag is pushed after opening the PR, the Action has to be re-run manually to pass.
There is no elegant way to have a workflow listen only to such tags onto PRs for specific branches...

**Comment by D-o-d-o-x** [comment]
This should probably be merged after #97. That way the Links on the PyPI-Page will also already be set to the final ones.

**Comment by D-o-d-o-x** [comment]
Also merge #99 before this one.

**Review by Onur4229 — APPROVED** [review]
